File: lib/src/test/java/net/mm2d/upnp/internal/message/FakeSsdpMessageTest.java
/*
 * Copyright (c) 2019 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.upnp.internal.message;

import net.mm2d.upnp.SsdpMessage;

import org.hamcrest.Matchers;
import org.junit.Before;
import org.junit.Test;

import java.io.OutputStream;
import java.net.InetAddress;

import static org.hamcrest.Matchers.*;
import static org.junit.Assert.*;
import static org.mockito.Mockito.*;

public class FakeSsdpMessageTest {
    private static final String LOCATION = "http://127.0.0.1/";
    private FakeSsdpMessage mFakeSsdpMessage;

    @Before
    public void setUp() {
        mFakeSsdpMessage = new FakeSsdpMessage(LOCATION);
    }

    @Test
    public void isPinned() {
        SsdpMessage message = new FakeSsdpMessage("", LOCATION, true);
        assertThat(message.isPinned(), is(true));
        message = new FakeSsdpMessage("", LOCATION, false);
        assertThat(message.isPinned(), is(false));
    }

    @Test
    public void getScopeId() {
        assertThat(mFakeSsdpMessage.getScopeId(), is(0));
    }

    @Test
    public void setLocalAddress() throws Exception {
        final InetAddress address = InetAddress.getByName("127.0.0.1");
        mFakeSsdpMessage.setLocalAddress(address);
        assertThat(mFakeSsdpMessage.getLocalAddress(), is(address));
    }

    @Test
    public void getLocalAddress() {
        assertThat(mFakeSsdpMessage.getLocalAddress(), is(nullValue()));
    }

    @Test
    public void getHeader() {
        assertThat(mFakeSsdpMessage.getHeader(""), is(nullValue()));
    }

    @Test
    public void setHeader() {
        mFakeSsdpMessage.setHeader("", "");
    }

    @Test
    public void setUuid() {
        final String uuid = "uuid";
        mFakeSsdpMessage.setUuid(uuid);
        assertThat(mFakeSsdpMessage.getUuid(), is(uuid));
    }

    @Test
    public void getUuid() {
        assertThat(mFakeSsdpMessage.getUuid(), isEmptyString());
    }

    @Test
    public void getType() {
        assertThat(mFakeSsdpMessage.getType(), isEmptyString());
    }

    @Test
    public void getNts() {
        assertThat(mFakeSsdpMessage.getNts(), Matchers.is(SsdpMessage.SSDP_ALIVE));
    }

    @Test
    public void getMaxAge() {
        assertThat(mFakeSsdpMessage.getMaxAge(), is(Integer.MAX_VALUE));
    }

    @Test
    public void getExpireTime() {
        assertThat(mFakeSsdpMessage.getExpireTime(), is(Long.MAX_VALUE));
    }

    @Test
    public void getLocation() {
        assertThat(mFakeSsdpMessage.getLocation(), is(LOCATION));
    }

    @Test
    public void writeData() {
        mFakeSsdpMessage.writeData(mock(OutputStream.class));
    }
}

File: mmupnp/src/test/java/net/mm2d/upnp/cp/internal/message/FakeSsdpMessageTest.kt
/*
 * Copyright (c) 2019 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.upnp.cp.internal.message

import com.google.common.truth.Truth.assertThat
import io.mockk.mockk
import net.mm2d.upnp.common.SsdpMessage
import org.junit.Before
import org.junit.Test

import java.net.InetAddress

class FakeSsdpMessageTest {
    private lateinit var fakeSsdpMessage: FakeSsdpMessage

    @Before
    fun setUp() {
        fakeSsdpMessage = FakeSsdpMessage(LOCATION)
    }

    @Test
    fun isPinned() {
        assertThat(FakeSsdpMessage(LOCATION, "", true).isPinned).isTrue()
        assertThat(FakeSsdpMessage(LOCATION, "", false).isPinned).isFalse()
    }

    @Test
    fun getScopeId() {
        assertThat(fakeSsdpMessage.scopeId).isEqualTo(0)
    }

    @Test
    fun setLocalAddress() {
        val address = InetAddress.getByName("127.0.0.1")
        fakeSsdpMessage.localAddress = address
        assertThat(fakeSsdpMessage.localAddress).isEqualTo(address)
    }

    @Test
    fun getLocalAddress() {
        assertThat(fakeSsdpMessage.localAddress).isNull()
    }

    @Test
    fun getHeader() {
        assertThat(fakeSsdpMessage.getHeader("")).isNull()
    }

    @Test
    fun setHeader() {
        fakeSsdpMessage.setHeader("", "")
    }

    @Test
    fun setUuid() {
        val uuid = "uuid"
        fakeSsdpMessage.uuid = uuid
        assertThat(fakeSsdpMessage.uuid).isEqualTo(uuid)
    }

    @Test
    fun getUuid() {
        assertThat(fakeSsdpMessage.uuid).isEmpty()
    }

    @Test
    fun getType() {
        assertThat(fakeSsdpMessage.type).isEmpty()
    }

    @Test
    fun getNts() {
        assertThat(fakeSsdpMessage.nts).isEqualTo(SsdpMessage.SSDP_ALIVE)
    }

    @Test
    fun getMaxAge() {
        assertThat(fakeSsdpMessage.maxAge).isEqualTo(Integer.MAX_VALUE)
    }

    @Test
    fun getExpireTime() {
        assertThat(fakeSsdpMessage.expireTime).isEqualTo(java.lang.Long.MAX_VALUE)
    }

    @Test
    fun getLocation() {
        assertThat(fakeSsdpMessage.location).isEqualTo(LOCATION)
    }

    @Test
    fun writeData() {
        fakeSsdpMessage.writeData(mockk(relaxed = true))
    }

    companion object {
        private const val LOCATION = "http://127.0.0.1/"
    }
}
